integrations/huggingface_hub/scripts/push_to_hub.py

```diff
@@ -0,0 +1,106 @@
+import os
+import typer
+import tempfile
+import tarfile
+import shutil
+import json
+from huggingface_hub import Repository, HfApi, HfFolder
+
+token_classification_components = ["ner", "tagger", "morphologizer"]
+text_classification_components = ["textcat", "textcat_multilabel"]
+
+
+def list_tags(name, tags):
+    if tags is None:
+        return ""
+    if isinstance(tags, str):
+        tags = [tags]
+    if len(tags) == 0:
+        return ""
+    name_tags = [f"- {tag}" for tag in tags]
+    return f"{name}:\n" + "\n".join(name_tags) + "\n"
+
+def create_model_card(repo_dir):
+    with open(os.path.join(repo_dir, "meta.json")) as f:
+        data = json.load(f)
+        lang = data["lang"] if data["lang"] != "xx" else "multilingual"
+        lic = data["license"].replace(" ", "-")
+        tags = ["spacy"]
+        for component in data["components"]:
+            if component in token_classification_components and "token-classification" not in tags:
+                tags.append("token-classification")
+            if component in text_classification_components and "text-classification" not in tags:
+                tags.append("text-classification")
+
+        metadata = list_tags("tags", tags)
+        metadata += list_tags("language", lang)
+        metadata += list_tags("license", lic)
+        metadata_section = f"---\n{metadata}---\n"
```

Comment:
would be super neat if you're able to extract the metrics from the model card and also store them in the `model-index` format (https://github.com/huggingface/huggingface_hub/blame/main/modelcard.md) which will let the hub filter models by a metric e.g. NER F1 on a given dataset, etc.

Reply:
I added support for the high level P/R/F metrics for NER, POS, dependencies and sentence segmentation. There are likely some metrics missing, I will double check.

It might also make sense to add the specific type metrics, but there is a lot of data there (see https://huggingface.co/spacy/en_core_web_sm/blob/main/en_core_web_sm-3.0.0/meta.json)